webapp/src/ts/modals/edit-user/edit-user-settings.component.ts

```diff
@@ -0,0 +1,119 @@
+import {Component, Input} from "@angular/core";
+import {EditUserAbstract} from "./edit-user.component";
+import {BsModalRef} from "ngx-bootstrap/modal";
+import {UserSettingsService} from "@mm-services/user-settings.service";
+import {UpdateUserService} from "@mm-services/update-user.service";
+import {LanguagesService} from "@mm-services/languages.service";
+
+@Component({
+  selector: 'update-password',
+  templateUrl: './edit-user-settings.component.html'
+})
+export class EditUserSettingsComponent extends EditUserAbstract {
+
+  @Input() editUserModel: {
+    id?,
+    fullname?,
+    username?,
+    email?,
+    place?,
+    contact?,
+    phone?,
+    language?: { code? }
+  } = {
+    language: {}
+  };
+
+  errors: any = {};
+  enabledLocales: any = [];
+
+  constructor(
+    bsModalRef: BsModalRef,
+    userSettingsService: UserSettingsService,
+    private updateUserService: UpdateUserService,
+    private languagesService: LanguagesService,
+  ) {
+    super(bsModalRef, userSettingsService);
+    this.languagesService.get().then((languages) => {
+      this.enabledLocales = languages;
+    });
+  }
+
+  editUserSettings(): Promise<void> {
+    this.setProcessing();
+    this.errors = {};
+    this.computeFields();
+
+    return this.changedUpdates(this.editUserModel).then((updates: any) => {
+      Promise.resolve().then(() => {
+        if (this.haveUpdates(updates)) {
+          return this.updateUserService.update(
+            this.editUserModel.username,
+            updates
+          ).toPromise();
+        }
+      })
+        .then(() => {
+          if (updates.language) {
+            // editing current user, so update language
+            // TODO once SetLanguage migrated refactor the commented code below
+            //SetLanguage(updates.language);
+          }
+          this.setFinished();
+          this.cancel();
+        })
+        .catch((err) => {
+          this.setError(err, 'Error updating user');
+        });
+    });
+  }
+
+  listTrackBy(index, locale) {
+    return locale.code;
+  }
+
+  private computeFields() {
+    if (document.querySelector('#edit-user-profile')) {
```

Comment:
Digging into this, I found there is a `div` with id `edit-user-profile` but in the user form in the _"App Management" > "Users"_ section at `/admin/#/users` where you edit a user as an admin user.

[admin/src/templates/edit_user.html](https://github.com/medic/cht-core/blob/a9ef90bb3edd2958ca913ba20e2e82c364696e2e/admin/src/templates/edit_user.html#L3):

```html

<mm-modal
  status="status"
  id="'edit-user-profile'"
```

The form is very similar to this, but is not the same code, and the other markup referenced in the code are also HTML elements from this form. Maybe at some point both forms were the same in the code, but then because the differences and because "admin" and "webapp" are kind of different apps, the code was copy and pasted but this part now is not necessary anymore here.

I'll remove the method `computeFields()` from the `EditUserSettingsComponent` component.